src/openrct2/ride/Vehicle.cpp

```diff
@@ -7269,6 +7276,50 @@ void Vehicle::UpdateSpinningCar()
     Invalidate();
 }
 
+void Vehicle::UpdateAnimationAnimalFlying()
+{
+    if (animationState > 0)
+    {
+        animationState--;
+        return;
+    }
+    else
+    {
+        auto trackType = GetTrackType();
+        TileElement* trackElement = map_get_track_element_at_of_type_seq(TrackLocation, trackType, 0);
+        // default waiting time is 3 ticks
+        // the timer determines the rest period for the next animation frame
+        // so case 3 setting animationState to 3 makes frame 0 play for 3 ticks
+        animationState = 3;
```

Comment:
I was thinking about this last night and wouldn't a table be even simpler:

```
constexpr std::array<uint16_t, 4> flyingStateWaitTime = { 3, 5, 3, 5 };
if (animationFrame == 0){
 // Insert animation start check, return if false...
animationState = flyingStateWaitTime[animationFrame];
} else {
animationState = flyingStateWaitTime[animationFrame];
//Move frame to next state code...
}
Invalidate();

Reply:
It will have to change to uint32_t as soon as the var_ca PR is merged. I don't think it adds anything to use the table other than making it look like paint code.

Reply:
One more reason, the current behavior catches out of bounds frames, something that the other animation types also do. A table doesn't do this so gracefully, and for only 4 valid states there's not much benefit.

Reply:
I basically don't want to see that fallthrough. It's bad design in general. Your wasting time computing the track element despite only using it for one state (0). You can get around any out of bounds issue if you modulo the lookup but it's not an issue as it's already bounded due to the &3. The size doesn't make any difference it can just as easily be a uint8 the types expand just fine to 32bit. Tbh my example it's a bit back to front slightly better would be:
```
if (animationFrame == 0){
 // Insert animation start check, return if false...
auto trackType = GetTrackType();
        TileElement* trackElement = map_get_track_element_at_of_type_seq(TrackLocation, trackType, 0);
      if (trackElement != nullptr && trackElement->AsTrack()->HasChain())
                {
                    // start flapping, bird
                    animation_frame = 1;
}
else{ return; }
} else {
// Next frame
animation_frame = (animation_frame + 1) % 4;
}
constexpr std::array<uint8_t, 4> flyingStateWaitTime = { 3, 5, 3, 5 };
animationState = flyingStateWaitTime[animationFrame];
Invalidate();
```
Sorry if it looks horrible doing this on mobile.